src/test/miner_tests.cpp

```diff
@@ -372,18 +372,44 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
     // Stop here if we needed to mine more blocks.
     assert(nblocks == sizeof(blockinfo)/sizeof(*blockinfo));
 
+    auto nTime = pblocktemplate->block.nTime;
+    delete pblocktemplate;
+
     // Set the clock to be just ahead of the last "mined" block, to ensure we satisfy the
     // future timestamp soft fork rule.
```

Comment:
This comment is now incorrect for the code it is associated with. Move this comment down to where the `-curTime + ...` logic now lives, and replace it here with an accurate comment for this part of the test.

Reply:
The comment is incorrect here because it says:
> Set the clock to be just ahead of the last "mined" block
but actually sets the clock to be _equal_ to the last "mined" block.

Reply:
> is almost a no-op by itself, apart from leaving the clock in fixed mode rather than offset mode.

AFAICT it is not a no-op. The call to `GetTime()` necessarily introduces some delay between itself and `nTime`, because the clock hasn't been overridden yet and is using the normal system clock. The new code has zero delay, per above.

Reply:
Yes. That was also incorrect prior to this change. I will open another PR to be applied after v6.0.0.

Reply:
Fixed by #6964.